message_sender/tasks.py

```diff
@@ -212,6 +209,13 @@ def run(self, message_id, **kwargs):
             l.info("Attempts: %s" % message.attempts)
             # send or resend
             try:
+                if not message.channel:
+                    channel = Channel.objects.get(default=True)
+                else:
+                    channel = Channel.objects.get(channel_id=message.channel)
```

Comment:
When we load the message above, we probably want to do a `select_related` for the channel, to avoid the additional database lookup (provided that we make the foreign key change)